guides/micronaut-graphql-todo/micronaut-graphql-todo.adoc

```diff
@@ -0,0 +1,398 @@
+
+[source,json]
+.Query
+----
+query {
+  toDos {
+    title,
+    completed,
+    author {
+       username
+    }
+  }
+}
+----
+
+Run the  following curl request:
+
+[source, bash]
+----
+curl -X POST 'http://localhost:8080/graphql' \
+     -H 'content-type: application/json' \
+     --data-binary '{"query":"{ toDos { title, completed, author { username } } }"}'
+----
+
+[source,json]
+----
+{"data":{"toDos":[]}}
+----
+
+We can then create a ToDo, by issuing a mutation query and return the ID of the newly created ToDo with the query:
+
+[source,json]
+.GraphQL Query
+----
+mutation {
+  createToDo(title: "Create GraphQL Guide", author: "Tim Yates") {
+    id
+  }
+}
+----
+
+Which translates to this Curl command:
+
+[source, bash]
+----
+curl -X POST 'http://localhost:8080/graphql' \
+     -H 'content-type: application/json' \
+     --data-binary '{"query":"mutation { createToDo(title:\"Create GraphQL Guide\", author:\"Tim Yates\") { id } }"}'
+----
+
+[source,json]
+.Response
+----
+{"data":{"createToDo":{"id":"1"}}}
+----
+
+This new ToDo then appears in our list of all ToDos with `completed` set to false:
+
+[source, bash]
+----
+curl -X POST 'http://localhost:8080/graphql' \
+     -H 'content-type: application/json' \
+     --data-binary '{"query":"{ toDos { title, completed, author { username } } }"}'
+----
+
+[source,json]
+.Response
+----
+{"data":{"toDos":[{"title":"Create GraphQL Guide","completed":false,"author":{"username":"Tim Yates"}}]}}
+----
+
+And we can mark it as completed by using this query with the ID from above (your ID will be different if you are following along)
+
+[source,json]
+.GraphQL query
+----
+mutation {
+  completeToDo(id: 1)
+}
+----
+
+[source,bash]
+----
+curl -X POST 'http://localhost:8080/graphql' \
+     -H 'content-type: application/json' \
+     --data-binary '{"query":"mutation { completeToDo(id: 1) }"}'
+----
+
+[source,json]
+.Response
+----
+{"data":{"completeToDo":true}}
+----
+
+And we can then see that this has been persisted in our in-memory model:
+
+[source,bash]
+.Query
+----
+curl -X POST 'http://localhost:8080/graphql' \
+     -H 'content-type: application/json' \
+     --data-binary '{"query":"{ toDos { title, completed } }"}'
+----
+
+[source,json]
+.Response
+----
+{"data":{"toDos":[{"title":"Create GraphQL Guide","completed":true}]}}
+----
+
+=== Automated tests
+
+For testing the application we will use Micronaut HTTP Client to send a `POST` request to the `/graphql` endpoint.
+Create the following class:
+
+test:GraphQLControllerTest[]
+
+When this test is executed, Testcontainers will start a fresh copy of Postgres in a container and apply all our migrations.
+The tests then run "as in production" with real data in a real database.
+
+To enable the tests to use this Dockerized database, we add a new resource `application-test.yml` to the test resources to overwrite the runtime datasource configuration.
```

Comment:
"we add a new resource application-test.yml to the test resources to overwrite" -> "create application-test.yml to overwrite"